### ISSUE TITLE
Default db directory on Android, migration from room

### ISSUE THREAD

**Issue by Viked** [tekartik/sqflite#67]
Hello, I need to use old db from Room, but getApplicationDocumentsDirectory returns not default database directory. 
The default path is //data/data/<Your-Application-Package-Name>/databases/<your-database-name>
[stackoverflow](https://stackoverflow.com/a/4452661/5526910)

For my app I have added this workaround

```
Directory documentsDirectory = await getApplicationDocumentsDirectory();
if (Platform.isAndroid) {
      Directory databaseDirectory =
          Directory(documentsDirectory.parent.path + '/databases');
      if (await databaseDirectory.exists() &&
          await File(databaseDirectory.path + '/' + dbName).exists()) {
        documentsDirectory = databaseDirectory;
      }
    }
```

**Comment by alextekartik** [comment]
Thanks for the note. Your current workaround looks fine to me. I noticed recently that `path_provider` seems to add an `app_flutter` part in the path.

Indeed I might need to add a way to get the default databases folder by adding an API in sqflite.

```dart
Future<Directory> getDatabasesDirectory();
```

What do you think?

On Android the proper way seems to create a database without an absolute path (juste a name) and get its path to get the database directory. (https://stackoverflow.com/questions/12723186/get-databases-directory-for-my-app-programatically) to avoid hardcoding `/databases` in the code.

On iOS I don't know if there is a specific folder or if people used the documents directory. We could maybe append `databases` as an opiniated location if people use `getDatabasesDirectory`. If someone else has a better alternative, I'm all ears.

**Comment by alextekartik** [comment]
I published a new version `0.11.0` with the added API:

```dart
Future<String> getDatabasesPath();
```
It is actually based on the `Context.getDatabasePath(String dbName)` API on Android and actually it avoids any `dart:io` reference in the API.
It returns the `databases` folder on Android and the `Documents` folder on iOS as you suggested.

By doing so I was able to remove the usage of `path_provider` in my example/test app. Let me know how it goes for you.
